packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/file.ts

```diff
@@ -0,0 +1,372 @@
+			joinLower = Math.min(member.joinedTimestamp, joinLower);
+			joinUpper = Math.max(member.joinedTimestamp, joinUpper);
+		}
+		creationLower = Math.min(member.user.createdTimestamp, creationLower);
+		creationUpper = Math.max(member.user.createdTimestamp, creationUpper);
+	}
+
+	const creationrange = ms(creationUpper - creationLower, true);
+	const joinrange = ms(joinUpper - joinLower, true);
+
+	await interaction.editReply({
+		content: `${i18next.t('command.mod.anti_raid_nuke.pending', {
+			members: members.size,
+			creationrange,
+			joinrange,
+			lng: locale,
+		})}\n\n${parameterStrings.join('\n')}`,
+		files: [{ name: `${potentialHitsDate}-anti-raid-nuke-list.txt`, attachment: potentialHits }],
+		components: [createMessageActionRow([cancelButton, banButton, dryRunButton])],
+	});
+
+	const collectedInteraction = await reply
+		.awaitMessageComponent({
+			filter: (collected) => collected.user.id === interaction.user.id,
+			componentType: ComponentType.Button,
+			time: 60000,
+		})
+		.catch(async () => {
+			try {
+				await interaction.editReply({
+					content: i18next.t('common.errors.timed_out', { lng: locale }),
+					components: [],
+				});
+			} catch (e) {
+				const error = e as Error;
+				logger.error(error, error.message);
+			}
+			return undefined;
+		});
+
+	if (collectedInteraction?.customId === cancelKey) {
+		await collectedInteraction.update({
+			content: i18next.t('command.mod.anti_raid_nuke.cancel', {
+				lng: locale,
+			}),
+			components: [],
+			attachments: [],
+		});
+	} else if (collectedInteraction?.customId === banKey || collectedInteraction?.customId === dryRunKey) {
+		const dryRunMode = collectedInteraction.customId === dryRunKey;
+
+		const content =
+			collectedInteraction.message.content +
+			(dryRunMode ? `\n\n${i18next.t('command.mod.anti_raid_nuke.parameters.dry_run', { lng: locale })}` : '');
+
+		await collectedInteraction.update({
+			content,
+			components: [
+				createMessageActionRow([
+					{ ...cancelButton, disabled: true },
+					{ ...banButton, disabled: true },
+					{ ...dryRunButton, disabled: true },
+				]),
+			],
+		});
+
+		await redis.setex(`guild:${collectedInteraction.guildId}:anti_raid_nuke`, 15, 'true');
+		let idx = 0;
+		const promises = [];
+
+		const result: AntiRaidResult[] = [];
+
+		for (const member of members.values()) {
+			promises.push(
+				(async () => {
+					const authorization = checkBan(member, interaction.user.id, ignoreRolesId);
+
+					if (authorization) {
+						result.push({
+							member,
+							success: false,
+							error: i18next.t(`command.mod.anti_raid_nuke.errors.result.${authorization}`, { lng: locale }),
+						});
+						return;
+					}
+
+					const reason = i18next.t('command.mod.anti_raid_nuke.reason', {
+						current: ++idx,
+						members: members.size,
+						lng: locale,
+					});
+
+					const ban = dryRunMode
+						? true
+						: await member.ban({ reason, deleteMessageDays: days }).catch((err) => {
```

Comment:
Not so much a fan of the whole await ().catch syntax, I rather have this as try/catch. also makes the nesting less deep.

Reply:
Implemented try catch on `[3efc23c](https://github.com/Naval-Base/yuudachi/pull/1069/commits/3efc23c48832b47a932d4eaf076d43a69ea1d958)`